### PR TITLE
ENH measure masked fraction

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds a new output, the masked fraction of the object. I am using a Gaussian aperture to measure this fraction. It is evaluated at the unsheared position.

### CODE REVIEW

File: metadetect/mfrac.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        obs = m.get_obs(i, 0)
+        wgt = obs.weight.copy()
+        msk = (obs.bmask & BMASK_EDGE) != 0
+        wgt[msk] = 0
+        wgt[~msk] = 1
+        obs.set_weight(wgt)
+
+        stats = gauss_wgt.get_weighted_sums(
+            obs,
+            obs.image.shape[0] * obs.jacobian.get_scale() / 2,
```
Comment (beckermr) [inline metadetect/mfrac.py:45]:
```suggestion
            fwhm * 2,
```

**Comment by beckermr** [comment]
@esheldon This is ready for review. I would consider writing a more detailed unittest if you think it is needed.

File: metadetect/mfrac.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import ngmix
+import numpy as np
+
+from .shearpos import DEFAULT_STEP
+from .detect import CatalogMEDSifier
+from .defaults import BMASK_EDGE
+
+
+def measure_mfrac(
```
Comment (esheldon) [inline metadetect/mfrac.py:9]: can you please add some docs here, and also explaining what it is you are doing?

File: metadetect/mfrac.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    gauss_wgt = ngmix.GMixModel(
+        [0, 0, 0, 0, ngmix.moments.fwhm_to_T(fwhm), 1],
+        'gauss',
+    )
+    mbobs = ngmix.MultiBandObsList()
+    obslist = ngmix.ObsList()
+    mbobs.append(obslist)
+    obslist.append(obs)
+    m = CatalogMEDSifier(mbobs, x, y, box_sizes).get_meds(0)
```
Comment (esheldon) [inline metadetect/mfrac.py:33]: why is this needed instead of our "usual" medsifier?

File: metadetect/metadetect.py [hunk 4]
```diff
@@ -231,6 +255,19 @@ def _add_positions_and_psf(self, cat, res, shear_str):
             else:
                 newres['ormask'] = self.ormask[rclip, cclip]
 
+            if np.any(self.mfrac > 0):
+                newres["mfrac"] = measure_mfrac(
+                    mfrac=self.mfrac,
+                    x=newres["sx_col_noshear"],
```
Comment (esheldon) [inline metadetect/metadetect.py:261]: In the PR  description you said you were measuring at the "unsheared position", is this what you meant?

File: metadetect/mfrac.py (hunk 3 above)
Comment (esheldon) [reply metadetect/mfrac.py:33]: what I mean is we already have a medsifier at this point right?  Why not re-use it?

File: metadetect/mfrac.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        msk = (obs.bmask & BMASK_EDGE) != 0
+        wgt[msk] = 0
+        wgt[~msk] = 1
+        obs.set_weight(wgt)
+
+        stats = gauss_wgt.get_weighted_sums(
+            obs,
+            fwhm * 2,
+        )
+        mfracs.append(stats["sums"][5] / stats["wsum"])
```
Comment (esheldon) [inline metadetect/mfrac.py:47]: comment: this gives the weighted average

**Review by esheldon — CHANGES_REQUESTED** [review]
Left some requests, questions and comments

File: metadetect/mfrac.py (hunk 2 above)
Comment (beckermr) [reply metadetect/mfrac.py:9]: 🤦 Sorry!

File: metadetect/metadetect.py (hunk 4 above)
Comment (beckermr) [reply metadetect/metadetect.py:261]: Yes.

File: metadetect/mfrac.py (hunk 5 above)
Comment (beckermr) [reply metadetect/mfrac.py:47]: Right. This is the masked pixel fraction weighted by the Gaussian aperture.

File: metadetect/mfrac.py (hunk 2 above)
Comment (beckermr) [reply metadetect/mfrac.py:8]: done!

**Comment by beckermr** [comment]
ok @esheldon ready for another look!

**Review by esheldon — APPROVED** [review]

**Comment by beckermr** [comment]
Awsome thanks!
